modules/GeneticsPortal.py

```diff
@@ -0,0 +1,534 @@
+            },
+            'mantissa': int(round(self.pval_mantissa)),
+            'exponent': int(self.pval_exponent),
+            'value': pval
+        }
+
+        # Generate evidence code field:
+        evidence_codes = [
+            self.evidence_code,
+            self.genetics_portal_base_url
+        ]
+
+        return_value = dict(gwas_sample_size=self.sample_size, provenance_type=self.generate_provenance(),
+                            is_associated=True,study_link='{}/study/{}'.format(self.genetics_portal_base_url, self.study_id),
+                            resource_score=resource_score, evidence_codes=evidence_codes, date_asserted=self.date_added,
+                            confidence_interval=self.confidence_interval, odds_ratio=self.odds_ratio)
+
+        # Adding reported trait:
+        return_value['reported_trait'] = self.reported_trait
+
+        # The literature link is only added if the pubmed ID is present:
+        return_value['unique_experiment_reference'] = '{}/{}'.format(self.literature_base_url, self.pmid) if self.pmid else '{}/0000'.format(self.literature_base_url)
+
+        return return_value
+
+    def generate_provenance(self):
+        # Generate provenence type:
+        provenence = {
+            'expert': {
+                'status': True,
+                'statement': 'Primary submitter of the data'
+            },
+            'database': {
+                'version': self.date_added,
+                'id': self.source_id,
+                'dbxref': {
+                    'version': self.date_added,
+                    'id': self.genetics_portal_base_url
+                }
+            }
+        }
+
+        if self.literature_prop:
+            provenence['literature'] = self.literature_prop
+
+        return provenence
+
+    def generate_gene2variant(self):
+
+        # Generate evidence code field:
+        evidence_codes = [
+            self.evidence_code,
+            self.genetics_portal_base_url
+        ]
+
+        # Generate resource score field:
+        resource_score = {
+            'type': 'locus_to_gene_score',
+            'method': {
+                'description': 'Locus to gene score generated by OpenTargets Genetics portal',
+            },
+            'value': self.l2g_score
+        }
+
+        return_data = {
+            'provenance_type': self.generate_provenance(),
+            'is_associated': True,
+            'date_asserted': self.date_added,
+            'evidence_codes': evidence_codes,
+            'resource_score': resource_score,
+            'functional_consequence': self.consequence_link,
+            'consequence_code': self.consequence_code
+        }
+
+        return return_data
+
+
+def initialize_evidence_generation(df):
+    """
+    As it is not possible for multiprocessing.Pool to copy over states of objects,
+    this function instantiates the evidence generator class and generates evidences.
+    """
+
+    global schemaFile
```

Comment:
the use of global is quite dangerous. I would say this case is not justified enough to bring a global value into a function used to map a collection. In that case, I would suggest the usage of [partial functions](https://docs.python.org/2/library/functools.html#functools.partial). A quick example to ilustrate the rewrite is drafted below

```python
>>> import functools as fn
>>> def func2(df, schema):
...     # df work
...     print schema
... 
>>> func1 = fn.partial(func2, schema="asdf")
>>> func1(42)
asdf

```

Reply:
As suggested, a partial function is used, so no global variable is used in the code. Thanks for the suggestion.